todo_app/data/session_items.py

```diff
@@ -67,3 +67,20 @@ def save_item(item):
     session['items'] = updated_items
 
     return item
+
+def delete_item(item):
+    """
+    removes item from the dictionary.
+
+    Args:
+        item: The item to remove.
+    """
+   
+    existing_items = get_items()
+    
+    item_id = item['id']-1
+    del existing_items[item_id]
```

Comment:
Your variable `item_id` is not an id, it's a guess at an index.

You are deleting the item at position `item['id'] - 1`. This fails if you delete an item from the middle of the list and then try to delete any other item

Quickest fix from this point is to get the actual index with `existing_items.index(item)`